deepchecks/nlp/checks/train_test_validation/property_drift.py

```diff
@@ -0,0 +1,270 @@
+        {feature_aggregation_method_argument:2*indent}
+    min_samples : int , default: 10
+        Minimum number of samples required to calculate the drift score. If there are not enough samples for either
+        train or test, the check will return None for that feature. If there are not enough samples for all properties,
+        the check will raise a ``NotEnoughSamplesError`` exception.
+    n_samples : int , default: 100_000
+        Number of samples to use for drift computation and plot.
+    random_state : int , default: 42
+        Random seed for sampling.
+    """
+
+    def __init__(
+        self,
+        properties: t.Union[t.Hashable, t.List[Hashable], None] = None,
+        ignore_properties: t.Union[Hashable, t.List[Hashable], None] = None,
+        n_top_properties: int = 5,
+        margin_quantile_filter: float = 0.025,
+        max_num_categories_for_drift: t.Optional[int] = None,
+        min_category_size_ratio: float = 0.01,
+        max_num_categories_for_display: int = 10,
+        show_categories_by: str = 'largest_difference',
+        numerical_drift_method: str = 'KS',
+        categorical_drift_method: str = 'cramers_v',
+        ignore_na: bool = True,
+        aggregation_method: t.Optional[str] = 'l3_weighted',
+        min_samples: int = 10,
+        n_samples: int = 100_000,
+        random_state: int = 42,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.properties = properties
+        self.ignore_properties = ignore_properties
+        self.n_top_properties = n_top_properties
+        self.margin_quantile_filter = margin_quantile_filter
+        self.max_num_categories_for_drift = max_num_categories_for_drift
+        self.min_category_size_ratio = min_category_size_ratio
+        self.max_num_categories_for_display = max_num_categories_for_display
+        self.show_categories_by = show_categories_by
+        self.numerical_drift_method = numerical_drift_method
+        self.categorical_drift_method = categorical_drift_method
+        self.ignore_na = ignore_na
+        self.aggregation_method = aggregation_method
+        self.min_samples = min_samples
+        self.n_samples = n_samples
+        self.random_state = random_state
+
+    def run_logic(self, context: Context) -> CheckResult:
+        """Run check."""
+        train = t.cast(TextData, context.train)
+        test = t.cast(TextData, context.test)
+
+        train_properties = select_from_dataframe(
+            train.properties,
+            columns=self.properties,
+            ignore_columns=self.ignore_properties
+        )
+        test_properties = select_from_dataframe(
+            test.properties,
+            columns=self.properties,
+            ignore_columns=self.ignore_properties
+        )
+
+        columns = set(train_properties.columns).intersection(set(test_properties.columns))
+        cat_columns = set([*train.categorical_properties, *test.categorical_properties])
+        results = {}
```

Comment:
All code below this, can they be put under a abstract class? Similar to what you did in the TrainTestPerformance

I mean here a common abstract class that will be the parent of this class and of FeatureDrift tabular check